content/spin/variables.md

```diff
@@ -0,0 +1,205 @@
+title = "Configuration Variables"
+template = "spin_main"
+date = "2022-06-14T00:22:56Z"
+enable_shortcodes = true
+[extra]
+url = "https://github.com/fermyon/developer/blob/main//content/spin/variables.md"
+
+---
+- [Using the Config SDK to Get Variables](#using-the-config-sdk-to-get-variables)
+
+Spin supports dynamic application variables. Instead of being static, their values can be updated without modifying the application, creating a simpler experience for rotating secrets, updating API endpoints, and more. 
+
+These variables are defined in a Spin application manifest (in the `[variables]` section) and are provided by a [configuration provider](/spin/dynamic-configuration.md#custom-config-providers). When running Spin locally, the configuration provider can be Vault for secrets or host environment variables. Refer to the [dynamic configuration documentation](/spin/dynamic-configuration.md) to learn how to configure variables locally.
+
+## Using the Config SDK to Get Variables
+
+The Spin SDK surfaces the Spin Config interface to your language. The [interface](https://github.com/fermyon/spin/blob/main/wit/ephemeral/spin-config.wit) consists of one operation:
+
+| Operation  | Parameters                          | Returns             | Behavior |
+|------------|-------------------------------------|---------------------|----------|
+| `get-config`    | Variable name  | Variable value    | Gets the value of the variable from the configured provider |
+
+In order to get a variable, it must be declared in an application's manifest (`spin.toml`). For example, say an application needs to access a secret. A variable named `password` could be added to the manifest and set as required with `required = true` since there is no reasonable default value for a secret. To do this, add a top-level `[variables]` section to the application manifest (`spin.toml`) and declare the variable within it.
```

Comment:
From an application authoring point of view, this feels a little backwards.  In order to _get_ a variable (application concern), it has to be declared on the component.  And then that harks back to an application variable, which is how you _set_ variables (operator concern).

This probably needs more explanation than can fit conveniently between "here is the underlying WIT API" and "here is how it gets surfaced in the SDKs".  Maybe a separate section?